lib/rules/padded-blocks.js

```diff
@@ -87,46 +115,87 @@ module.exports = function(context) {
-
-            if (requirePadding) {
-                if (!blockHasTopPadding) {
-                    context.report(node, ALWAYS_MESSAGE);
-                }
-                if (!blockHasBottomPadding) {
-                    context.report({
-                        node: node,
-                        loc: {line: node.loc.end.line, column: node.loc.end.column - 1 },
-                        message: ALWAYS_MESSAGE
-                    });
-                }
-            } else {
-                if (blockHasTopPadding) {
-                    context.report(node, NEVER_MESSAGE);
-                }
-
-                if (blockHasBottomPadding) {
-                    context.report({
-                        node: node,
-                        loc: {line: node.loc.end.line, column: node.loc.end.column - 1 },
-                        message: NEVER_MESSAGE
-                    });
-                }
+        var openBrace = getOpenBrace(node),
+            closeBrace = sourceCode.getLastToken(node),
+            blockHasTopPadding = isTokenTopPadded(openBrace),
+            blockHasBottomPadding = isTokenBottomPadded(closeBrace);
+
+        if (requirePaddingFor(node)) {
+            if (!blockHasTopPadding) {
+                context.report({
+                    node: node,
+                    loc: { line: openBrace.loc.start.line, column: openBrace.loc.start.column },
+                    message: ALWAYS_MESSAGE
+                });
+            }
+            if (!blockHasBottomPadding) {
+                context.report({
+                    node: node,
+                    loc: {line: closeBrace.loc.end.line, column: closeBrace.loc.end.column - 1 },
+                    message: ALWAYS_MESSAGE
+                });
+            }
+        } else {
+            if (blockHasTopPadding) {
+                context.report({
+                    node: node,
+                    loc: { line: openBrace.loc.start.line, column: openBrace.loc.start.column },
+                    message: NEVER_MESSAGE
+                });
+            }
+
+            if (blockHasBottomPadding) {
+                context.report({
+                    node: node,
+                    loc: {line: closeBrace.loc.end.line, column: closeBrace.loc.end.column - 1 },
+                    message: NEVER_MESSAGE
+                });
             }
         }
     }
 
     return {
-        "BlockStatement": checkPadding
+        "SwitchStatement": function(node) {
+            if (node.cases.length === 0) {
+                return;
+            }
+            if (!options.hasOwnProperty("switches")) {
+                return;
+            }
+            checkPadding(node);
+        },
+        "BlockStatement": function(node) {
+            if (node.body.length === 0) {
+                return;
+            }
+            if (!options.hasOwnProperty("blocks")) {
+                return;
+            }
+            checkPadding(node);
+        }
```

Comment:
Hey! Just had an idea for a free performance boost :fireworks:

Since we know what the options are before we return the traversal callback, if an option isn't configured, we don't even need to be notified when the corresponding node type is encountered during a traversal! Rearrange this a bit so `"BlockStatement"`s are only included in the returned callback if `blocks` are configured, same for `SwitchStatement`, and that'll save a few cycles.